### DISCUSSION TITLE
Deduplication and rules.

### DISCUSSION THREAD

**Discussion by pmobbe** [firefly-iii/firefly-iii#4441] [Firefly III] [answered]
CSV Importer v.2.5.0
Firefly III v. 5.4.6

I am using the CSV importer to import bank statements into Firefly and in general that works great. One thing that I am struggling with is to get deduplication to work. I have it set to "content" deduplication strategy but yet when I import the same  file twice I get duplicate records. 

Can this be because I have rules that are triggered during the import? Are rules applied before or after the deduplication?

**Comment by JC5** [comment]
Rules don't influence the deduplication process, so that's not it.

Importing the same file twice doesn't lead to duplicates. What kind of duplicates are you getting exactly? Have you [compared](https://jsoncompare.org/) the debug view?

https://docs.firefly-iii.org/csv/errors/duplicate_transactions/

**Comment by pmobbe** [comment, marked as answer]
Just wanted to write back and let you know that I ran another test and I am no longer having the issue where duplicates are being created. Not really sure what I did wrong the first time but subsequent tries have been successful.

**Reply by JC5** [reply]
Thanks for letting me know, I was worried what it could be. Keep me posted in case it breaks in the future.
